walker.go

```diff
@@ -161,6 +162,14 @@ func Walk(ctx context.Context, p string, opt *WalkOpt, fn filepath.WalkFunc) err
 			}
 
 			if isDir {
+				fi, err := dirEntry.Info()
+				if err != nil {
+					if isNotExist(err) {
```

Comment:
This extra `if` is not needed here. Already handled in defer.

Reply:
oops, my mistake, i misunderstood your comment above